packages/shared/src/porter.ts

```diff
@@ -284,4 +326,48 @@ export class PorterClient {
     > = Object.fromEntries(decryptionResponses);
     return { encryptedResponses, errors };
   }
-}
+
+
+  public async signUserOp(
+    signingRequests: Record<string, string>,
+    threshold: number,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    _options: SigningOptions = { optimistic: true, returnAggregated: true },
+  ): Promise<SignResult> {
+    const data: Record<string, unknown> = {
+      signing_requests: signingRequests,
+      threshold: threshold
+    };
+    
+    const resp: AxiosResponse<SignResponse> = await this.tryAndCall({
+      url: '/sign',
+      method: 'post',
+      data,
+    });
+
+    const { signatures, errors } = resp.data.result.signing_results;
+    
+    const signingResults: { [ursulaAddress: string]: [string, string] } = {};
+    let messageHash = '';
+
+    // Process signatures - decode the JSON payload to extract message hash and signature
+    for (const [ursulaAddress, [signerAddress, signatureB64]] of Object.entries(signatures || {})) {
+      const decodedData = JSON.parse(
+        new TextDecoder().decode(fromBase64(signatureB64))
+      );
+      messageHash = decodedData.message_hash;
+      signingResults[ursulaAddress] = [signerAddress, decodedData.signature];
+    }
+
+    const aggregatedSignature = Object.keys(signingResults).length > 0 
```

Comment:
We should only provide an aggregate id there are sufficient signatures i.e. a threshold of them.
```suggestion
    const aggregatedSignature = Object.keys(signingResults).length > threshold 
```

Reply:
^ @theref